upload-api/tables/delegations.js

```diff
@@ -134,27 +151,54 @@ async function cidToDelegation (bucket, cid) {
   return delegation
 }
 
-/**
- * 
- * @param {import('../types').DelegationsBucket} bucket
- * @param {Ucanto.Delegation<Ucanto.Tuple<Ucanto.Capability>>[]} delegations
+/** 
+ * @param {import('../types').InvocationBucket} invocationBucket
+ * @param {import('../types').WorkflowBucket} workflowBucket
+ * @param {CID} invocationCid
+ * @param {CID} delegationCid
+ * @returns {Promise<Ucanto.Delegation>}
  */
-async function writeDelegations (bucket, delegations) {
-  return writeEntries(
-    bucket,
-    [...delegations].map((delegation) => {
-      const carBytes = delegationsToBytes([delegation])
-      const value = carBytes
-      return /** @type {[key: CID, value: Uint8Array]} */ ([delegation.cid, value])
-    })
+async function findDelegationInInvocation (invocationBucket, workflowBucket, invocationCid, delegationCid) {
+
+  // TODO is this right? I cargo culted some code from ucan-invocation.js and w3up/upload-api/src/access/delegate.js
+  // but am not highly confident it works
+
+  const invocationCidStr = invocationCid.toString()
+  const agentMessageWithInvocationCid = await invocationBucket.getInLink(
+    invocationCid.toString()
   )
-}
 
-/**
- * 
- * @param {import('../types').DelegationsBucket} bucket
- * @param {Iterable<readonly [key: CID, value: Uint8Array ]>} entries
- */
-async function writeEntries (bucket, entries) {
-  await Promise.all([...entries].map(([key, value]) => bucket.put(key, value)))
+  if (!agentMessageWithInvocationCid) {
+    throw new NoCarFoundForGivenReceiptError()
```

Comment:
Does this trigger on a receipt? I think we need a different error. This error in `ucan-invocation` is because the flow is:
1. process received CAR file with receipt
2. get CAR file from invocation that corresponds do the receipt, so that we have the execution parameters of the invocation that generated the receipt.

Therefore, here error should be `NoCarFoundForGivenDelegation` or similar.

Reply:
Done! Introduced a couple new errors that feel like they better reflect what's going on. I could break down the `NoInvocationFoundForGivenCidError` error into a couple more precise errors about different parts of the invocation lookup process, but wanted to start simple.